blocks/logic.js

```diff
@@ -20,54 +20,290 @@
+  },
+  // Block for comparison operator.
+  {
+    "type": "logic_compare",
+    "message0": "%1 %2 %3",
+    "args0": [
+      {
+        "type": "input_value",
+        "name": "A"
+      },
+      {
+        "type": "field_dropdown",
+        "name": "OP",
+        "options": [
+          ["=", "EQ"],
+          ["\u2260", "NEQ"],
+          ["<", "LT"],
+          ["\u2264", "LTE"],
+          [">", "GT"],
+          ["\u2265", "GTE"]
+        ]
+      },
+      {
+        "type": "input_value",
+        "name": "B"
+      }
+    ],
+    "inputsInline": true,
+    "output": "Boolean",
+    "colour": "%{BKY_LOGIC_HUE}",
+    "helpUrl": "%{BKY_LOGIC_COMPARE_HELPURL}",
+    "extensions": ["logic_compare", "logic_op_tooltip"]
+  },
+  // Block for logical operations: 'and', 'or'.
+  {
+    "type": "logic_operation",
+    "message0": "%1 %2 %3",
+    "args0": [
+      {
+        "type": "input_value",
+        "name": "A",
+        "check": "Boolean"
+      },
+      {
+        "type": "field_dropdown",
+        "name": "OP",
+        "options": [
+          ["%{BKY_LOGIC_OPERATION_AND}", "AND"],
+          ["%{BKY_LOGIC_OPERATION_OR}", "OR"]
+        ]
+      },
+      {
+        "type": "input_value",
+        "name": "B",
+        "check": "Boolean"
+      }
+    ],
+    "inputsInline": true,
+    "output": "Boolean",
+    "colour": "%{BKY_LOGIC_HUE}",
+    "helpUrl": "%{BKY_LOGIC_OPERATION_HELPURL}",
+    "extensions": ["logic_op_tooltip"]
+  },
+  // Block for negation.
+  {
+    "type": "logic_negate",
+    "message0": "%{BKY_LOGIC_NEGATE_TITLE}",
+    "args0": [
+      {
+        "type": "input_value",
+        "name": "BOOL",
+        "check": "Boolean"
+      }
+    ],
+    "output": "Boolean",
+    "colour": "%{BKY_LOGIC_HUE}",
+    "tooltip": "%{BKY_LOGIC_NEGATE_TOOLTIP}",
+    "helpUrl": "%{BKY_LOGIC_NEGATE_HELPURL}"
+  },
+  // Block for null data type.
+  {
+    "type": "logic_null",
+    "message0": "%{BKY_LOGIC_NULL}",
+    "output": "null",
+    "colour": "%{BKY_LOGIC_HUE}",
+    "tooltip": "%{BKY_LOGIC_NULL_TOOLTIP}",
+    "helpUrl": "%{BKY_LOGIC_NULL_HELPURL}"
+  },
+  // Block for ternary operator.
+  {
+    "type": "logic_ternary",
+    "message0": "%{BKY_LOGIC_TERNARY_CONDITION} %1",
+    "args0": [
+      {
+        "type": "input_value",
+        "name": "IF",
+        "check": "Boolean"
+      }
+    ],
+    "message1": "%{BKY_LOGIC_TERNARY_IF_TRUE} %1",
+    "args1": [
+      {
+        "type": "input_value",
+        "name": "THEN"
+      }
+    ],
+    "message2": "%{BKY_LOGIC_TERNARY_IF_FALSE} %1",
+    "args2": [
+      {
+        "type": "input_value",
+        "name": "ELSE"
+      }
+    ],
+    "output": true,
+    "colour": "%{BKY_LOGIC_HUE}",
+    "tooltip": "%{BKY_LOGIC_TERNARY_TOOLTIP}",
+    "helpUrl": "%{BKY_LOGIC_TERNARY_HELPURL}",
+    "extensions": ["logic_ternary"]
+  }
+]);  // END JSON EXTRACT (Do not delete this comment.)
+
+Blockly.defineBlocksWithJsonArray([ // Mutator blocks. Do not extract.
```

Comment:
Note the lack of "BEGIN JSON EXTRACT" is intentional.

Note that I've now added the beginning marker to distinguish it from sections like this set of mutator blocks (useless to Android and iOS).